### ISSUE TITLE
[QUERY][Azure.Core] Validate resource Id with ResourceIdentifier

### ISSUE THREAD

**Issue by karataliu** [Azure/azure-sdk-for-net#28988]
### Library name and version

Azure.Core 1.24.0

### Query/Question

What is the correct way to verify a string is valid resource ID?

current we need to 
```csharp
var rid = new ResourceIdentifier(input);
var _ = rid.Name; // Accessing one property verify FormatException
```

Is there a canonical way to do this?

**Comment by jsquire** [comment]
//cc: @m-nash

**Comment by m-nash** [comment]
This is a good question.  The `ResourceIdentifier` type was meant to lazy parse the string to optimize performance on deserialization when there might be hundreds of these in a given response and the user probably won't touch all of them.  Right now the only way to "test" a string would be to construct then access a property and catch.

We can probably add an API to allow a user to test a string, will talk with the team and ping back.

**Comment by m-nash** [comment]
@karataliu I will mention that creating ResourceIdentifiers isn't that easy so we have added helper methods to all mgmt plane libraries to construct the strings for you from the parts.  [This section of the readme](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/README.md#structured-resource-identifier) talks about it briefly.  For example if you want to create a VirtualMachine ResourceIdentifier you can call [this method here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/compute/Azure.ResourceManager.Compute/src/Generated/VirtualMachineResource.cs#L31) and it prompts you for the pieces that are needed as well as constructs the string for you.

Basically any Resource type in Azure has this helper method to construct these for you to avoid creating invalid strings accidentally.
